### PR TITLE
Apply umesh's patch to mod.rs

### PR DESCRIPTION
**Pull request by BaoshanPang**
r? @n-salim

### CODE REVIEW

File: src/vxworks/mod.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Remove this blank line.

File: src/vxworks/mod.rs [hunk 2]
```diff
@@ -36,25 +78,31 @@ pub type ssize_t = isize;
...
-pub type ino32_t = u32;
 pub type off_t = ::c_longlong;
 
-pub type rlim_t = ::c_ulonglong;
+pub type rlim_t = ::c_ulong;
 pub type suseconds_t = ::c_long;
 pub type time_t = ::c_long;
-pub type wchar_t = ::c_int;
+
+cfg_if! {
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Good catch (Umesh). But we have arch specific files in libc, maybe wchar_t should go there?

File: src/vxworks/mod.rs [hunk 3]
```diff
@@ -68,24 +116,20 @@ pub type uid_t = ::c_ushort;
...
 
 // From b_off_t.h
 pub type off64_t = ::c_longlong;
-pub type off_t64 = ::c_longlong;
 
 // From b_BOOL.h
 pub type BOOL = ::c_int;
 
-// From vxWind.h ..
+//From vxWind.h ..
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Why was the space removed? The previous formatting was better.

File: src/vxworks/mod.rs [hunk 4]
```diff
@@ -132,23 +173,25 @@ s! {
...
-        pub condSemName: [::c_char; PTHREAD_SHARED_SEM_NAME_MAX]
+        pub condSemName: [::c_char; _PTHREAD_SHARED_SEM_NAME_MAX]
     }
 
     // b_pthread_rwlockattr_t.h
     pub struct pthread_rwlockattr_t {
         pub rwlockAttrStatus: ::c_int,
+	pub rwlockAttrPshared: ::c_int,
         pub rwlockAttrMaxReaders: ::c_uint,
+	pub rwlockAttrConformOpt: ::c_uint,
```
Comment (n-salim) [inline src/vxworks/mod.rs:174]: Very important catches! The missing fields would have led to stack corruption when using R/W locks.

File: src/vxworks/mod.rs [hunk 5]
```diff
@@ -132,23 +173,25 @@ s! {
...
     // b_pthread_rwlock_t.h
     pub struct pthread_rwlock_t {
         pub rwlockSemId: :: _Vx_SEM_ID,
-        pub rwlockReadersRefCount: ::c_int,
+        pub rwlockReadersRefCount: ::c_uint,
         pub rwlockValid: ::c_int,
         pub rwlockInitted: ::c_int,
         pub rwlockAttr: ::pthread_rwlockattr_t,
-        pub rwlockName: [::c_char; PTHREAD_SHARED_SEM_NAME_MAX]
+        pub rwlockSemName: [::c_char; 30]
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Prefer to use named constant, so long as it has the correct value.

File: src/vxworks/mod.rs [hunk 6]
```diff
@@ -175,17 +218,17 @@ s! {
         pub ss_family  : ::sa_family_t,
         pub __ss_pad1  : [::c_char; _SS_PAD1SIZE],
         pub __ss_align : i32,
-        // pub __ss_pad2  : [::c_char; _SS_PAD2SIZE],
-        pub __ss_pad2  : [::c_char; 32],
-        pub __ss_pad3  : [::c_char; 32],
-        pub __ss_pad4  : [::c_char; 32],
-        pub __ss_pad5  : [::c_char; _SS_PAD2SIZE - 96],
+        pub __ss_pad2  : [::c_char; _SS_PAD2SIZE],
+        //pub __ss_pad2  : [::c_char; 32],
```
Comment (n-salim) [inline src/vxworks/mod.rs]: No need for the commented out lines. Just remove them. We can always find them again by looking at older versions if needed.

File: src/vxworks/mod.rs [hunk 7]
```diff
@@ -212,37 +255,21 @@ s! {
     // dirent.h
     pub struct dirent {
         pub d_ino  : ::ino_t,
-        // pub d_name : [::c_char; (_PARM_NAME_MAX + 1)],
-        pub d_name : [::c_char; 32],
+        pub d_name : [::c_char; 256],
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Named constant better than magic number

File: src/vxworks/mod.rs [hunk 8]
```diff
@@ -212,37 +255,21 @@ s! {
...
+        /*pub d_name : [::c_char; 32],
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Just remove these lines.

File: src/vxworks/mod.rs [hunk 9]
```diff
@@ -985,18 +998,19 @@ pub const SEEK_END: ::c_int = 2;
 pub const VX_RTP_NAME_LENGTH: usize = 255;
 
 //Some unsupported stuff
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Not sure this comment is helpful. Just remove it?

File: src/vxworks/mod.rs [hunk 10]
```diff
@@ -1128,7 +1142,7 @@ extern {
         n: size_t,
     ) -> c_int;
     pub fn strlen(cs: *const c_char) -> size_t;
-    pub fn strnlen(cs: *const c_char, maxlen: size_t) -> size_t;
+    //pub fn strnlen(cs: *const c_char, maxlen: size_t) -> size_t;
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Just remove it if we don't have it.

File: src/vxworks/mod.rs [hunk 11]
```diff
@@ -1439,6 +1453,7 @@ extern {
 
     // pthread.h
     pub fn pthread_mutexattr_init(
+        /* PTHREAD STUFF */
```
Comment (n-salim) [inline src/vxworks/mod.rs]: Comment is redundant. Pretty obvious that this is "pthread stuff"!

File: src/vxworks/mod.rs (hunk 2 above)
Comment (UmeshKalappa) [reply src/vxworks/mod.rs]: Agree

File: src/vxworks/mod.rs [hunk 12]
```diff
@@ -526,7 +536,7 @@ pub const PTHREAD_MUTEX_ERRORCHECK: ::c_int = 1;
 pub const PTHREAD_MUTEX_RECURSIVE: ::c_int = 2;
 pub const PTHREAD_MUTEX_DEFAULT: ::c_int = PTHREAD_MUTEX_NORMAL;
 pub const PTHREAD_STACK_MIN: usize = 4096;
-pub const PTHREAD_SHARED_SEM_NAME_MAX: usize = 30;
+pub const _PTHREAD_SHARED_SEM_NAME_MAX: usize = 30;
```
Comment (BaoshanPang) [inline src/vxworks/mod.rs:513]: Why we add "_" here?

**Comment by UmeshKalappa** [comment]
To match vxworks definition.

**Review by n-salim — APPROVED** [review]

**Pull request merged**